lib/core/CStaticThreadPool.cc

```diff
@@ -65,18 +67,30 @@ void CStaticThreadPool::busy(bool value) {
 }
 
 void CStaticThreadPool::shutdown() {
+
+    // Wait here until all the queues are empty. This is to ensure that each thread
+    // is waiting to pop its own queue. This is critical because each thread must
+    // execute its own shutdown message.
+    auto notIsEmpty = [](TTaskQueue& queue) { return queue.size() > 0; };
+    while (std::find_if(m_TaskQueues.begin(), m_TaskQueues.end(), notIsEmpty) !=
+           m_TaskQueues.end()) {
+        std::this_thread::sleep_for(std::chrono::microseconds{50});
```

Comment:
This seems like a code smell to me.  We could commit it as a temporary fix to stop CI breaking, but I imagine there is a pattern for shutting down threadpools that doesn't involve sleeping.

Reply:
I thought about this a bit...

There is if we modify the queue. I could have special case handling in the queue which allows me to pass a message to notify all consumer conditions (blocked on pop on empty queue) to wake up and exit.

I kind of like the idea of waiting until we've drained the queues. It means we don't complicate the queue and we don't have any extra logic going on all the time whenever we pop or push onto the queue. I don't think we'll need to start and stop thread pools for our use case. I could probably do this cleaner with a condition variable, but this was a quick fix for the tests on the branch. I'll leave a TODO to tidy this up.